provider/marathon_test.go

```diff
@@ -336,27 +339,37 @@ func TestMarathonLoadConfig(t *testing.T) {
 	}
 
 	for _, c := range cases {
-		fakeClient := newFakeClient(c.applicationsError, c.applications, c.tasksError, c.tasks)
-		provider := &Marathon{
-			Domain:           "docker.localhost",
-			ExposedByDefault: true,
-			marathonClient:   fakeClient,
+		appID := ""
+		if len(c.applications.Apps) > 0 {
+			appID = c.applications.Apps[0].ID
 		}
-		actualConfig := provider.loadMarathonConfig()
-		fakeClient.AssertExpectations(t)
-		if c.expectedNil {
-			if actualConfig != nil {
-				t.Fatalf("Should have been nil, got %v", actualConfig)
-			}
-		} else {
-			// Compare backends
-			if !reflect.DeepEqual(actualConfig.Backends, c.expectedBackends) {
-				t.Fatalf("expected %#v, got %#v", c.expectedBackends, actualConfig.Backends)
+		t.Run(fmt.Sprintf("Running case: %s", appID), func(t *testing.T) {
+			fakeClient := newFakeClient(c.applicationsError, c.applications, c.tasksError, c.tasks)
+			provider := &Marathon{
+				Domain:           "docker.localhost",
+				ExposedByDefault: true,
+				marathonClient:   fakeClient,
 			}
-			if !reflect.DeepEqual(actualConfig.Frontends, c.expectedFrontends) {
-				t.Fatalf("expected %#v, got %#v", c.expectedFrontends, actualConfig.Frontends)
+			actualConfig := provider.loadMarathonConfig()
+			fakeClient.AssertExpectations(t)
+			if c.expectedNil {
+				if actualConfig != nil {
+					t.Fatalf("Should have been nil, got %v", actualConfig)
+				}
+			} else {
+				// Compare backends
+				if !reflect.DeepEqual(actualConfig.Backends, c.expectedBackends) {
+					expected, _ := json.Marshal(c.expectedBackends)
+					actual, _ := json.Marshal(actualConfig.Backends)
+					t.Fatalf("expected\t %s\n, \tgot %s\n", expected, actual)
```

Comment:
This should be an `t.Errorf` IMHO so that we can still see the frontend assertion even if the backend one fails.